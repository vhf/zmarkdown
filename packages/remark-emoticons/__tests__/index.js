import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import plugin from '../src/'

const render = text => unified()
  .use(reParse)
  .use(plugin, {
    ':ange:': '/static/smileys/ange.png',
    ':colere:': '/static/smileys/angry.gif',
    'o_O': '/static/smileys/blink.gif',
    ';)': '/static/smileys/clin.png',
    ':diable:': '/static/smileys/diable.png',
    ':D': '/static/smileys/heureux.png',
    '^^': '/static/smileys/hihi.png',
    ':o': '/static/smileys/huh.png',
    ':p': '/static/smileys/langue.png',
    ':magicien:': '/static/smileys/magicien.png',
    ':colere2:': '/static/smileys/mechant.png',
    ':ninja:': '/static/smileys/ninja.png',
    'x(': '/static/smileys/pinch.png',
    ':pirate:': '/static/smileys/pirate.png',
    ":'(": '/static/smileys/pleure.png',
    ':lol:': '/static/smileys/rire.gif',
    ':honte:': '/static/smileys/rouge.png',
    ':-Â°': '/static/smileys/siffle.png',
    ':)': '/static/smileys/smile.png',
    ':soleil:': '/static/smileys/soleil.png',
    ':(': '/static/smileys/triste.png',
    ':euh:': '/static/smileys/unsure.gif',
    ':waw:': '/static/smileys/waw.png',
    ':zorro:': '/static/smileys/zorro.png',
  })
  .use(remark2rehype)
  .use(stringify)
  .processSync(text)

test('emoticons', () => {
  const { contents } = render(dedent`
    Hello :) Hey :D

    :)

    > Citation

    :D This is not a caption

    ![toto](https://zestedesavoir.com/media/galleries/3014/bee33fae-2216-463a-8b85-d1d9efe2c374.png)

    :D This is not a caption

    This is not a smiley:)

    Not :)a smiley either.

    Smiley after another node: [link](#foo) :)

    Smiley after another node w/2 spaces: [link](#foo)  :)

    Smiley after another node w/3 spaces: [link](#foo)   :)
  `)
  expect(contents).toMatchSnapshot()
})

test('Errors without config', () => {
  const fail = () => unified()
    .use(reParse)
    .use(remark2rehype)
    .use(plugin)
    .use(stringify)
    .processSync('')

  expect(fail).toThrowError(Error)
})
