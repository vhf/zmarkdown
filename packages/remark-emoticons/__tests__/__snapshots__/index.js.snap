// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emoticons 1`] = `
"<p>Hello <img src=\\"/static/smileys/smile.png\\" alt=\\":)\\" class=\\"foo bar\\"> Hey <img src=\\"/static/smileys/heureux.png\\" alt=\\":D\\" class=\\"foo bar\\"></p>
<p><img src=\\"/static/smileys/smile.png\\" alt=\\":)\\" class=\\"foo bar\\"></p>
<blockquote>
<p>Citation</p>
</blockquote>
<p><img src=\\"/static/smileys/heureux.png\\" alt=\\":D\\" class=\\"foo bar\\"> This is not a caption</p>
<p><img src=\\"https://zestedesavoir.com/media/galleries/3014/bee33fae-2216-463a-8b85-d1d9efe2c374.png\\" alt=\\"toto\\"></p>
<p><img src=\\"/static/smileys/heureux.png\\" alt=\\":D\\" class=\\"foo bar\\"> This is not a caption</p>
<p>This is not a smiley:)</p>
<p>Not :)a smiley either.</p>
<p>Smiley after another node: <a href=\\"#foo\\">link</a><img src=\\"/static/smileys/smile.png\\" alt=\\":)\\" class=\\"foo bar\\"></p>
<p>Smiley after another node w/2 spaces: <a href=\\"#foo\\">link</a>  <img src=\\"/static/smileys/smile.png\\" alt=\\":)\\" class=\\"foo bar\\"></p>
<p>Smiley after another node w/3 spaces: <a href=\\"#foo\\">link</a>   <img src=\\"/static/smileys/smile.png\\" alt=\\":)\\" class=\\"foo bar\\"></p>"
`;

exports[`emoticons without class 1`] = `
"<p>Hello <img src=\\"/static/smileys/smile.png\\" alt=\\":)\\"> Hey <img src=\\"/static/smileys/heureux.png\\" alt=\\":D\\"></p>
<p><img src=\\"/static/smileys/smile.png\\" alt=\\":)\\"></p>
<blockquote>
<p>Citation</p>
</blockquote>
<p><img src=\\"/static/smileys/heureux.png\\" alt=\\":D\\"> This is not a caption</p>
<p><img src=\\"https://zestedesavoir.com/media/galleries/3014/bee33fae-2216-463a-8b85-d1d9efe2c374.png\\" alt=\\"toto\\"></p>
<p><img src=\\"/static/smileys/heureux.png\\" alt=\\":D\\"> This is not a caption</p>
<p>This is not a smiley:)</p>
<p>Not :)a smiley either.</p>
<p>Smiley after another node: <a href=\\"#foo\\">link</a><img src=\\"/static/smileys/smile.png\\" alt=\\":)\\"></p>
<p>Smiley after another node w/2 spaces: <a href=\\"#foo\\">link</a>  <img src=\\"/static/smileys/smile.png\\" alt=\\":)\\"></p>
<p>Smiley after another node w/3 spaces: <a href=\\"#foo\\">link</a>   <img src=\\"/static/smileys/smile.png\\" alt=\\":)\\"></p>"
`;

exports[`renders case insensitive 1`] = `
"<p><img src=\\"/static/smileys/langue.png\\" alt=\\":p\\"> is same as <img src=\\"/static/smileys/langue.png\\" alt=\\":P\\"></p>
<p><img src=\\"/static/smileys/heureux.png\\" alt=\\":d\\"> is same as <img src=\\"/static/smileys/heureux.png\\" alt=\\":D\\"></p>
<p><img src=\\"/static/smileys/blink.gif\\" alt=\\"o_O\\"> is same as <img alt=\\"o_o\\"> and <img alt=\\"O_O\\"></p>
<p><img src=\\"/static/smileys/magicien.png\\" alt=\\":magicien:\\"> is same as <img src=\\"/static/smileys/magicien.png\\" alt=\\":MAGICIEN:\\"> and as <img src=\\"/static/smileys/magicien.png\\" alt=\\":mAgIcIeN:\\"></p>"
`;

exports[`renders to markdown 1`] = `
"Hello :) Hey :D

:)

> Citation

:D This is not a caption

![toto](https://zestedesavoir.com/media/galleries/3014/bee33fae-2216-463a-8b85-d1d9efe2c374.png)

:D This is not a caption

This is not a smiley:)

Not :)a smiley either.

Smiley after another node: [link](#foo):)

Smiley after another node w/2 spaces: [link](#foo)  :)

Smiley after another node w/3 spaces: [link](#foo)   :)
"
`;
